mesonbuild/build.py

```diff
@@ -137,6 +137,12 @@ def get_target_macos_dylib_install_name(ld) -> str:
     name.append('.dylib')
     return ''.join(name)
 
+
+def extract_targets_as_list(kwargs: T.Dict[str, T.Any], key: str, self_libs: T.List[LibTypes]) -> T.List[LibTypes]:
+    lib_list = listify(kwargs.get(key, [])) + self_libs
+    return [t.get_default_object() if isinstance(t, BothLibraries) else t for t in lib_list]
```

Comment:
I'm not thrilled with this because the use of `Any` here in the `dict[str, Any]` will prevent the type checker from helping us out, as opposed to just:
```python
def targets_as_list(extra_libs: T.Sequence[LibTypes], self_libs: T.Sequence[LibTypes]) -> T.List[LibTypes]:
    return [t.get_default_object() if isinstance(t, BothLibraries) else t
               for t in itertools.chain(extra_libs, self_libs)]
```

and being used as:
```python
x = targets_as_list(kwargs.get('link_with', []), self.link_targets)
```
This means that if `kwargs` is annotated, that the type checker can see into the function.